client/src/app/services/chart.service.ts

```diff
@@ -525,13 +540,60 @@ export class ChartService {
   }
   //#endregion
 
+  //#region WINDOW AND TIMEFRAME OPERATIONS
+  
+  onWindowResize(dimensions: { width: number; height: number }) {
+    const newBarCount = this.window.calculateOptimalBars(dimensions.width);
+    
+    // Only update if bar count changed significantly and we have data
+    if (Math.abs(newBarCount - this.currentBarCount) > 10 && this.allQuotes.length > 0) {
+      this.currentBarCount = newBarCount;
+      this.updateChartsWithNewBarCount();
+    }
+  }
+  
+  private updateChartsWithNewBarCount() {
+    // Slice existing data to new bar count (keep newest data)
+    const quotes = this.allQuotes.slice(-this.currentBarCount);
+    
+    console.log(`Updating charts with ${this.currentBarCount} bars (dynamic resize)`);
+    
+    // Update main chart with new data slice
+    this.loadOverlayChart(quotes);
```

Comment:
Double check to ensure this set of operations dos not change the quantity or configuration of those user choices.  For example it should not add extra selections than before resizing.